src/steps/devices/index.ts

```diff
@@ -492,6 +482,19 @@ export async function fetchComputers({
   const computerDeviceIdToGraphObjectKeyMap: DeviceIdToGraphObjectKeyMap =
     new Map();
 
+  await iterateComputerGroups(client, async (computerGroup) => {
+    {
+      let computerGroupEntity = await jobState.findEntity(
+        generateEntityKey(Entities.COMPUTER_GROUP._type, computerGroup.name),
+      );
+      if (computerGroupEntity === null) {
+        computerGroupEntity = await jobState.addEntity(
+          createComputerGroupEntity(computerGroup.name),
+        );
+      }
+    }
+  });
+
```

Comment:
```suggestion
  await iterateComputerGroups(client, async (computerGroup) => {
    const hasComputerKey = jobState.hasKey(
      generateEntityKey(Entities.COMPUTER_GROUP._type, computerGroup.name),
    );
    if (!hasComputerKey) {
      await jobState.addEntity(createComputerGroupEntity(computerGroup.name));
    }
  });

```

I would recommend using `hasKey` over `findEntity` as often as possible. hasKey is more performant. Please note this recommendation: @suraj-metron

Also, the code has an additional set of `{}`. Intentional?
And on line 491, the `computerGroupEntity` is being set without being used again. Intentional?

Reply:
removing computerGroupEntity as it is not used